Server/src/RedisManager.cpp

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2024 Microsoft, Inc. and others.  All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ */
+
+#include "RedisManager.h"
+
+
+/*********************************************************************//**
+ * Constructor for class
+ ***********************************************************************/
+RedisManager::RedisManager() : stateDb_(BMP_DB_NAME, 0, true) {
+    swss::SonicDBConfig::initialize();
+    swss::SonicDBConfig::initializeGlobalConfig();
+    separator_ = swss::SonicDBConfig::getSeparator(BMP_DB_NAME);
+    exit_ = false;
+}
+
+/*********************************************************************//**
+ * Constructor for class
+ ***********************************************************************/
+RedisManager::~RedisManager() {
+    if (!exit_) {
+        exit_ = true;
+        for (auto& threadPtr : threadList_) {
+            threadPtr->join();
+        }
+    }
+}
+
+
+/*********************************************************************
+ * Setup logger for this class
+ *
+ * \param [in] logPtr     logger pointer
+ ***********************************************************************/
+void RedisManager::Setup(Logger *logPtr, BMPListener::ClientInfo *client) {
+    logger = logPtr;
+    client_ = client;
+}
+
+
+
+/**
+ * Get Key separator for deletion
+ *
+ * \param [in] N/A
+ */
+std::string RedisManager::GetKeySeparator() {
+    return separator_;
+}
+
+
+/**
+ * WriteBMPTable
+ *
+ * \param [in] table            Reference to table name
+ * \param [in] key              Reference to various keys list
+ * \param [in] fieldValues      Reference to field-value pairs
+ */
+bool RedisManager::WriteBMPTable(const std::string& table, const std::vector<std::string>& keys, const std::vector<swss::FieldValueTuple> fieldValues) {
+
+    if (enabledTables_.find(table) == enabledTables_.end()) {
+        LOG_INFO("RedisManager %s is disabled", table.c_str());
+        return false;
+    }
+
+    swss::Table stateBMPTable(&stateDb_, table);
+    std::string fullKey;
+    for (const auto& key : keys) {
+        fullKey += key;
+        fullKey += separator_;
```

Comment:
>separator_ [](http://example.com/codeflow?start=19&length=10)

separator_ is implementation details of Table class, and should be hidden. I do not understand why you need to manipulate. Did you check the raw data you wrote to database?